pkg/handlers/handlers.go

```diff
@@ -422,7 +426,13 @@ func handleChunksWithError(
 			}
 			if len(dataOrErr.Data) > 0 {
 				chunk := *chunkSkel
+				if chunk.SourceMetadata != nil {
+					if cloned, ok := proto.Clone(chunk.SourceMetadata).(*source_metadatapb.MetaData); ok {
+						chunk.SourceMetadata = cloned
```

Comment:
I'm not sure what this is doing, but for various personal reasons I'm behind on sleep 😹 so it's likely I'm missing something obvious